### ISSUE TITLE
Block Gauss-Seidel test fails when cuSPARSE is enabled

### ISSUE THREAD

**Issue by brian-kelley** [kokkos/kokkos-kernels#648]
Reported by @lucbv: building with: ``--with-tpls=cusparse``, the block Gauss-Seidel unit test fails because of mismatched matrix/vector size in SPMV:
```
[ RUN      ] cuda.sparse_block_gauss_seidel_rank1_double_int_int_TestExecSpace
 ** On entry to cusparseSpMV() dimension mismatch: matA->cols != vecX->size
unknown file: Failure
C++ exception with description "KokkosSparse::spmv[TPL_CUSPARSE,double]: cusparseSpMV() failed" thrown in the test body.
[  FAILED  ] cuda.sparse_block_gauss_seidel_rank1_double_int_int_TestExecSpace (735 ms)
```

This suggests 2 things:
- Our SPMV implementation isn't checking that dimensions match, allowing this test to pass without cuSPARSE
- The block Gauss-Seidel test itself is generating the matrix and/or LHS vector with the wrong dimensions

**Comment by brian-kelley** [comment]
Hmm, our spmv implementation does check the input dimensions, but it only checks that the matrix dimensions are not _greater_ than the corresponding x/y lengths. So the matrix can have fewer columns than the dimension of X, and fewer rows than the dimension of Y. I assume this was intentional, but if it wasn't, I can fix it.

The issue in block Gauss-Seidel is clear, though:
```
[ RUN      ] cuda.sparse_block_gauss_seidel_rank1_double_int_int_TestExecSpace
Calling spmv (rank-1), with m = 2002, n = 2002, xlen = 2009, ylen = 2002
 ** On entry to cusparseSpMV() dimension mismatch: matA->cols != vecX->size
```
The x vector is clearly too long by 7 elements (which happens to be the block_size). Should be easy to fix there.

**Comment by brian-kelley** [comment]
Yep, it was an instance of #478, sort of:
```
lno_t nv = ((crsmat2.numRows() / block_size)+1) * block_size;
```
should be:
```
lno_t nv = ((crsmat2.numRows() + block_size - 1) / block_size) * block_size;
```
and now it passes with cusparse.

**Comment by lucbv** [comment]
@brian-kelley thanks for looking into it, we can probably raise this issue tomorrow.
I am a bit worried about the test we perform in SPMV but if that is a design decision we can probably live with it.

**Comment by brian-kelley** [comment]
Fixed with #649.
